src/daft-sql/src/modules/image.rs

```diff
@@ -1,11 +1,92 @@
+use daft_dsl::{Expr, ExprRef, LiteralValue};
+use sqlparser::ast::{FunctionArg, FunctionArgOperator};
+
 use super::SQLModule;
-use crate::functions::SQLFunctions;
+use crate::{
+    error::SQLPlannerResult,
+    functions::{FromSQLArgs, SQLFunction, SQLFunctions},
+    unsupported_sql_err,
+};
+use daft_functions::image::decode::{decode, ImageDecode};
 
 pub struct SQLModuleImage;
 
 impl SQLModule for SQLModuleImage {
-    fn register(_parent: &mut SQLFunctions) {
-        // use FunctionExpr::Image as f;
-        // TODO
+    fn register(parent: &mut SQLFunctions) {
+        parent.add_fn("image_decode", SQLImageDecode {});
+    }
+}
+
+pub struct SQLImageDecode;
+
+impl FromSQLArgs for ImageDecode {
+    fn from_sql(
+        inputs: &[sqlparser::ast::FunctionArg],
+        planner: &crate::planner::SQLPlanner,
+    ) -> SQLPlannerResult<Self> {
+        let mut mode = None;
+        let mut raise_on_error = true;
+        for arg in inputs {
```

Comment:
WDYT of a utility that would take `inputs` and produce a `(named_args: HashMap<&str, Expr>, unnamed_args: HashMap<usize, Expr>)`?

That way we can also easily handle things such as users providing bad inputs like providing names twice, or providing unnamed args after named args etc

Reply:
I was initially thinking of doing something like this, but didn't want to overcomplicate things. 

Will update!

Reply:
@jaychia I have this updated in the other PR https://github.com/Eventual-Inc/Daft/pull/2764